addons/survey/models/survey_survey.py

```diff
@@ -106,7 +172,7 @@ def copy_data(self, default=None):
         return super(Survey, self).copy_data(default)
 
     @api.multi
-    def _create_answer(self, user=False, partner=False, email=False, test_entry=False, **additional_vals):
+    def _create_answer(self, user=False, partner=False, email=False, test_entry=False, attempts_pool_token=False, **additional_vals):
```

Comment:
To discuss and clean; as this method is more about can I create an answer than am I allowed to create an answer
 * we have an invite_token -> check attempts
 * we don't have an invite_token -> create one if survey has attempt limited

Routes and controllers would have to check the invite only mode linked to limited attempts and then call create answer.

cc @jem-odoo

Reply:
That method already checks a lot of things when calling "_check_answer_creation" so I thought it would be easier to centralize all the checks there (including the attempts check).

As we've changed this method about 10 times already I'm not sure if I still need to move the check or not?

Reply:
(Done as discussed with another parameter "check_attempts")